src/pages/TripBookingPage/TripBookingPage.tsx

```diff
@@ -1,54 +1,129 @@
 import PassengerDetailsForm from './PassengerDetailsForm/PassengerDetailsForm';
 import LongArrow from '../../components/icons/LongArrow';
-import { TripCardAccordion } from '../../components';
-import { ITrip, ISeatType, IBusType } from '../../types';
+import { ConfirmDialog, TripCardAccordion } from '../../components';
 import { useAppSelector } from '../../app/hooks';
+import { StyledButton } from '../../components/Button/Button.styled';
+import { FareDetails } from '../../components/FareDetails/FareDetails';
+import { useEffect, useRef, useState } from 'react';
+import { FormikProps } from 'formik';
+import { IPassengersInputFromFormik, filterSelectedSeats } from '../../utils';
+import { useNavigate } from 'react-router-dom';
+import { toast } from 'react-toastify';
+import { useTranslation } from 'react-i18next';
+import FullScreenLoader from '../../components/FullScreenLoader/FullScreenLoader';
+import { createSelector } from '@reduxjs/toolkit';
+import { RootState } from '../../app/store';
 import { fromSerializable } from '../../utils/tripDetailsUtils';
-
 // TODO: fetch data from store
-const dummyTripData: ITrip = {
-    id: '1',
-    origin: {
-        name: 'Chennai',
-        id: '1',
-        shortCode: 'MAA',
-    },
-    destination: {
-        name: 'Bangalore',
-        id: '2',
-        shortCode: 'BLR',
-    },
-    departureTimestamp: new Date('2024-02-01T06:00:00Z'),
-    arrivalTimestamp: new Date('2024-02-01T12:00:00Z'),
-    seatType: ISeatType.SLEEPER,
-    busType: IBusType.AC,
-    farePerSeat: 50,
-    availableSeats: 20,
-    totalSeats: 30,
-};
 
 export const TripBookingPage = () => {
-    const state = useAppSelector((state) =>
-        fromSerializable(state.tripDetails)
+    const { t } = useTranslation('bookingPageConfirmation');
+    const navigate = useNavigate();
+    const isMinWidth = useMediaQuery('(min-width:600px)');
+
+    const selectTripDetails = (state: RootState) => state.tripDetails;
+    const selectSerializedTripDetails = createSelector(
+        selectTripDetails,
+        fromSerializable
     );
+    const state = useAppSelector(selectSerializedTripDetails);
+
+    const selectedSeats = filterSelectedSeats(state.seats);
+    const selectedSeatsCount = selectedSeats.length;
+
+    const [showDialog, setShowDialog] = useState(false);
+    const [showLoader, setShowLoader] = useState(false);
+    const setShowLoaderFun = (bool: boolean) => {
+        setShowLoader(bool);
+    };
+    const formikRef = useRef<FormikProps<IPassengersInputFromFormik>>(null);
+
+    const handleFormSubmit = () => {
+        if (formikRef.current?.isValid) {
+            setShowDialog(true);
+        } else {
+            formikRef.current?.handleSubmit(); // to retrigger validation of the form
+        }
+    };
+    useEffect(() => {
+        if (state.seats.length === 0) {
+            navigate(-1);
+            toast.error(t('absentStateError'));
```

Comment:
noAvailableSeatError ?

Reply:
This is the condition when the store doesn't have the total current booking data. This happens when the user reloads the booking page or directly navigates to the booking page without selecting any seats.